lisa/util/shell.py

```diff
@@ -119,7 +119,11 @@ def _initialize(self) -> None:
         try:
             connect(paramiko_client, self._connection_info)
         except Exception as identifier:
-            raise LisaException(f"connect to server failed: {identifier}")
+            raise LisaException(
+                f"connect to server "
+                f"[{self._connection_info.address}:{self._connection_info.port}]"
+                f" failed: {identifier}"
```

Comment:
I have never used f-strings before and these 3 lines here have convinced me that I was missing out on something so good. 👍

Reply:
> Changes LGTM from a cursory glance, but I'm not versed in the Azure APIs or that JSON DSL to do a more thorough review.
> 
> I'd appreciate if you could split your changes into multiple commits if possible; most of your changes touches multiple files and are hard to reason about. Have you ever used `git add -p` before?

I will try `git add -p` next time, and split into mutiple commits.